core/src/nodes/collection.rs

```diff
@@ -189,11 +197,11 @@ impl<'a, TInEvent, TOutEvent, THandler, TReachErr, THandlerErr> CollectionReachE
     }
 
     /// Accepts the new node.
-    pub fn accept(self) -> (CollectionNodeAccept, PeerId) {
+    pub fn accept(self) -> (CollectionNodeAccept, TPeerId) {
         // Set the state of the task to `Connected`.
         let former_task_id = self.parent.nodes.insert(self.peer_id.clone(), self.id);
         let _former_state = self.parent.tasks.insert(self.id, TaskState::Connected(self.peer_id.clone()));
-        debug_assert_eq!(_former_state, Some(TaskState::Pending));
+        debug_assert!(_former_state == Some(TaskState::Pending));
```

Comment:
Why the change from `debug_assert_eq` to `debug_assert` here?

Reply:
Otherwise we have to require `Debug` on the `TPeerId`.

Reply:
Is this not required here? https://github.com/libp2p/rust-libp2p/pull/881/files#diff-87276c9dd75246456f7330545b24dd12R128
```rust
       TPeerId: Eq + Hash + Clone + fmt::Debug,
```